src/emulator/io/include/io/device.h

```diff
@@ -0,0 +1,141 @@
+// Vita3K emulator project
+// Copyright (C) 2018 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#pragma once
+
+#include <io/VitaIoDevice.h>
+
+#include <util/fs.h>
+
+#include <string>
+
+namespace device {
+
+/**
+  * \brief Get a Vita device from a given path.
+  * \param path The input path to be tested.
+  * \return The path's root as an enumeration. Otherwise, VitaIoDevice::_INVALID.
+  */
+inline VitaIoDevice get_device(const std::string &path) {
```

Comment:
See above resolved with pent0.

Some functions were (later) moved for the compiler.

Reply:
Ahh, are you talking about https://github.com/Vita3K/Vita3K/pull/505#discussion_r305556235? Why only some and not all?

Reply:
Part lessening clutter, part class definition.

I treated the `device` namespace as a pseudo-class for `VitaIoDevice` (like member functions). This pretty much meant having short, compiler-friendly, fast functions as `inline`, and having them as such will be beneficial in the long run since they are faster (e.g., having `get_device` as inline will be really helpful in the long run due to all of the device path logic it does).

Reply:
I mean, the cost of calling the function is going to be just a few cycles, plus I don't know how often they are really going to be called. I don't think they really need to be inline imo. Up to you though. If you think it's good, it's good.